### PR TITLE
[Page] Normalise spacing within Header

### PR DESCRIPTION
**Pull request by mrcthms**
### WHY are these changes introduced?

Addresses https://github.com/Shopify/shopify/issues/322184

### WHAT is this pull request doing?

This PR introduces a universal 20px spacing at the top and bottom of the Header within the `Page` component. It also tightens up the spacing between a Title and Subtitle – if one is present – from 8px to 4px, as well as ensuring that the Title is visually centered with any other elements within the main Header content area.

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, ping @ sarahill to update the Polaris UI kit

### CODE REVIEW

File: src/components/Page/Page.tsx [hunk 1]
```diff
@@ -38,6 +38,7 @@ export function Page({
 
   const contentClassName = classNames(
     styles.Content,
+    !hasHeaderContent && styles.ContentSpaced,
```
Comment (alex-page) [inline src/components/Page/Page.tsx]: Above on line 35 `.Content` is renamed to `.ContentSpaced` should we remove `styles.Content`?

File: src/components/Page/tests/Page.test.tsx [hunk 2]
```diff
@@ -271,10 +271,10 @@ describe('<Page />', () => {
     it('does not render border when divider is true and no header props exist', () => {
       const wrapper = mountWithApp(<Page divider />);
       expect(wrapper).not.toContainReactComponent('div', {
-        className: 'Content divider',
+        className: 'Content ContentSpaced divider',
```
Comment (kyledurand) [inline src/components/Page/tests/Page.test.tsx]:
```suggestion
        className: 'ContentSpaced divider',
```

File: src/components/Page/tests/Page.test.tsx [hunk 3]
```diff
@@ -271,10 +271,10 @@ describe('<Page />', () => {
...
       });
       expect(wrapper).toContainReactComponent('div', {
-        className: 'Content',
+        className: 'Content ContentSpaced',
```
Comment (kyledurand) [inline src/components/Page/tests/Page.test.tsx]:
```suggestion
        className: 'ContentSpaced',
```

**Review by kyledurand — APPROVED** [review]
Code looks fine. I'd prefer to stick with the `.Content` classname though. `.ContentSpaced` sounds like it's a modifier on `Content` when it's really not

**Pull request merged**